### ISSUE TITLE
Inserting data into REAL column in PostgreSQL table via Presto results in wrong data in table

### ISSUE THREAD

**Issue by losipiuk** [prestodb/presto#8635]
To reproduce:

* configure Presto to connect PSQL instance

* on PSQL:
```
psql> create table ftestpg(a real, d double precision);
```

* on Presto:
```
presto> insert into ftestpg values (real '1.0', 1.0);
presto> select * from ftestpg;
      a       |  d
--------------+-----
 1.06535322E9 | 1.0
```

The reason for that behaviour is that decision how insert is performed to SQL connector is made in many cases just based on 
Presto container type (and not actual SQL type) in `JDBCRecordSink`. In case of `REAL` `appendLong` method is called, which results in
`statement.setLong` being called on prepared statement while `setFloat` should have been.
There is high chance that the problem is present also for some other SQL types.
The code of `append*` methods should explicitly check for SQL type of handled column, and perform logic based on it. The default branch should
just throw exception "Unsupported SQL type".

**Comment by findepi** [comment]
Gonna be fixed by #9393

**Comment by electrum** [comment]
This should be fixed now
